pkg/signature/kms/cliplugin/options.go

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cliplugin implements the plugin functionality.
+package cliplugin
+
+import (
+	"context"
+	"crypto"
+	"time"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/kms/cliplugin/common"
+)
+
+// packRPCOptions extracts properties of all of opts into struct ready for serializing.
+func packRPCOptions(opts []signature.RPCOption) *common.RPCOptions {
```

Comment:
Could these functions be in the same file as the unpack.*Options?

Reply:
we could, but it could lead to a circular import situation in the future.

Reply:
In what way? This effectively is marshal and unmarshal, so grouping them together seems logically structured.

Reply:
When I was trying to think of a way for the kms package to import the cliplugin package directly, I thought the handler package might need to depend on the clipluign package, and maybe causing a circular dependency. But that won't be the case, so this isn't an issue.

I moved the pack and unpack functions to a new encoding package.